### PR TITLE
Add sonarcloud yaml workflows

### LINKED ISSUE #12: Missing SonarCloud Scanner
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

There is no code scanner right now installed in the repository.

### Describe the solution you'd like

Install SonarCloud using Github actions

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by axeltaylor**
# Add missing SonarCloud Scanner to workflows

Add sonarcloud for dotnet API and Angular web project

Fixes #12

### CODE REVIEW

File: .github/workflows/sonarcloud-api.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          dotnet tool update dotnet-sonarscanner --tool-path .\.sonar\scanner
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN_API }}
+        shell: powershell
+        run: |
+          .\.sonar\scanner\dotnet-sonarscanner begin /k:"starterkit-angular-and-dotnet-api_api" /o:"erni-academy" /d:sonar.login="${{ secrets.SONAR_TOKEN_API }}" /d:sonar.host.url="https://sonarcloud.io" /d:sonar.dotnet.excludeTestProjects="true"
+          dotnet restore
+          dotnet build --configuration Release --no-restore
```
Comment (Rabosa616) [inline .github/workflows/sonarcloud-api.yaml:53]: You need to set where is the .sln file 
dotnet build ./src/App.API/App.Api.sln --configuration Release --no-restore

File: .github/workflows/sonarcloud-web.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          required: true
+  workflow_dispatch: ~
+jobs:
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
```
Comment (Rabosa616) [inline .github/workflows/sonarcloud-web.yaml:17]: I miss the part of the  npm run-script build:prod
Also is missing the file sonar-project.properties file

File: .github/workflows/sonarcloud-web.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
```
Comment (Rabosa616) [inline .github/workflows/sonarcloud-web.yaml:19]: Please check the folder where runs the step, it should be ./src/App.UI

File: .github/workflows/sonarcloud-api.yaml (hunk 1 above)
Comment (axeltaylor) [reply .github/workflows/sonarcloud-api.yaml]: Done

File: .github/workflows/sonarcloud-web.yaml (hunk 2 above)
Comment (axeltaylor) [reply .github/workflows/sonarcloud-web.yaml:20]: Done

File: .github/workflows/sonarcloud-web.yaml (hunk 3 above)
Comment (axeltaylor) [reply .github/workflows/sonarcloud-web.yaml:27]: Done

**Review by Rabosa616 — APPROVED** [review]

**Pull request merged**